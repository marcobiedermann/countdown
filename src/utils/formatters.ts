function formatTime(value: number) {
  return Math.abs(value).toString().padStart(2, '0');
}

export { formatTime };
