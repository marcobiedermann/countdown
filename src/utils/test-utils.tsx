import { render as rtlRender } from '@testing-library/react';
import i18n from 'i18next';
import type { ReactNode } from 'react';
import { I18nextProvider, initReactI18next } from 'react-i18next';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';

import deTranslation from '../../public/locales/de/translation.json';
import enTranslation from '../../public/locales/en/translation.json';
import { options } from '../i18n';
import { store } from '../store';

i18n.use(initReactI18next).init({
  ...options,
  resources: {
    de: {
      translation: deTranslation,
    },
    en: {
      translation: enTranslation,
    },
  },
});

interface WrapperProps {
  children: ReactNode;
}

function render(ui: ReactNode, { route = '/' } = {}) {
  function Wrapper(props: WrapperProps) {
    const { children } = props;

    return (
      <Provider store={store}>
        <I18nextProvider i18n={i18n}>
          <MemoryRouter initialEntries={[route]}>{children}</MemoryRouter>
        </I18nextProvider>
      </Provider>
    );
  }

  return {
    ...rtlRender(ui, { wrapper: Wrapper }),
  };
}

export { render };
