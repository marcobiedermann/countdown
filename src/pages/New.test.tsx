import { describe, expect, it } from 'vitest';
import { render } from '../utils/test-utils';
import NewPage from './New';

describe('NewPage', () => {
  it('should render correctly', () => {
    const { container } = render(<NewPage />);

    expect(container).toMatchSnapshot();
  });
});
