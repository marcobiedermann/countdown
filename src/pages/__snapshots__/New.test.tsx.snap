// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`NewPage > should render correctly 1`] = `
<div>
  <form
    class="form"
  >
    <div
      class="form__field"
    >
      <label
        class="form__label"
        for="title"
      >
        Title
      </label>
      <input
        class="form__input"
        id="title"
        name="title"
        type="text"
      />
    </div>
    <div
      class="form__field"
    >
      <label
        class="form__label"
        for="date"
      >
        Date
      </label>
      <div
        class="react-datepicker-wrapper"
      >
        <div
          class="react-datepicker__input-container react-datepicker__view-calendar-icon"
        >
          <svg
            class=" react-datepicker__calendar-icon undefined false"
            fill="none"
            height="1em"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <rect
              height="18"
              rx="2"
              ry="2"
              width="18"
              x="3"
              y="4"
            />
            <line
              x1="16"
              x2="16"
              y1="2"
              y2="6"
            />
            <line
              x1="8"
              x2="8"
              y1="2"
              y2="6"
            />
            <line
              x1="3"
              x2="21"
              y1="10"
              y2="10"
            />
          </svg>
          <input
            class="form__input"
            type="text"
            value=""
          />
        </div>
      </div>
    </div>
    <button
      class="button button--primary button--pill"
      type="submit"
    >
      Create
    </button>
  </form>
</div>
`;
